docs/proposals/optional_application_loging.md

```diff
@@ -0,0 +1,144 @@
+Optional Application Logging
+============================
+
+The EFK aggregated logging stack collects application logs from STDOUT of each 
+container managed by OpenShift using fluentd. Currently, only way of opting out
+is based on a node label and ansible variable 
+`openshift_logging_fluentd_nodeselector` as described in the
+[documentation](https://docs.openshift.org/latest/install_config/aggregate_logging.html).
+The cluster operator can deploy aggregated logging setting the
+`openshift_logging_fluentd_nodeselector`, which controls on what nodes are the 
+fluentd pods deployed, and container logs are only collected on those nodes
+where fluentd pod is deployed.
+
+This proposal discusses some ways of a finer grained level of control over what 
+container logs are processed and stored using the EFK stack.
+
+## Fluentd pipeline
+
+The container logs can be obtained from one of these supported logging drivers
+- json-file - logs read from `/var/log/containers/*` mounted in fluentd
+              container
+- journald
+
+Latest development shifted recommended docker driver from journald to json-file,
+therefore, the proposed implementation has focused on usage of json-file as
+well. Rationale being, journald currently doesn't match performance expectations
+and desired throttling control for larger clusters.
+
+Each container on the host has its own json structured log file
+- `/var/log/containers/[pod]_[namespace]_[container]-[docker_id].log`
+
+Every line is treated as a separate log message and processed in the cascade
+of fluentd filters. A log line doesn't contain OpenShift metadata, this is 
+added to the message in the fluentd `k8_metadata_filter` plugin.
+
+#### Dynamic Control of Optional Logging
+
+By 'dynamic control' we understand the ability of an operator or a user to
+opt-in or opt-out certain OpenShift object from aggregated logging without
+having to redeploy fluentd. This approach has immediate effect, offers very 
+granular level of control, is flexible, and a change of settings in order
+to disable logging on a particular container doesn't introduce a spike in 
+latency for other containers logs.
+
+It requires additional resources:
+- memory - cache `O(count(namespaces)+count(pods))` + process overhead
+- CPU - parsing notices + process overhead
+
+and also brings overhead to the API server, where the fluentd input plugin has 
+to `watch` `namespaces` and `pods` API endpoints.
+
+Following solution describes how this can be implemented as an opt-out strategy:
+- a user labels a namespace or a pod with `aggregated_logging_opt_out=true`
```

Comment:
shouldn't the label (or annotation) be something more like `io.openshift.logging=true`

Reply:
of course, that is open for discussion. I selected a string that fell unique and difficult to get into a collision with any other component, but I like your suggestion better. Do we have a document with label/annotation naming convention so I comply with it in my future proposals?